core/blockchain/impl/block_tree_impl.cpp

```diff
@@ -246,6 +281,21 @@ namespace kagome::blockchain {
     metric_known_chain_leaves_->set(tree_->getMetadata().leaves.size());
   }
 
+  const primitives::BlockHash &BlockTreeImpl::getGenesisBlockHash() const {
+    if (genesis_block_hash_.has_value()) {
+      return genesis_block_hash_.value();
+    }
+
+    auto res = header_repo_->getHashByNumber(0);
+    BOOST_ASSERT_MSG(res.has_value(),
+                     "Block tree must contain at least genesis block");
+
+    const_cast<std::decay_t<decltype(genesis_block_hash_)> &>(
```

Comment:
It just alternative way. const_cast has selected because it a single point to change. Otherwise, the mutable member might be changed in any method silently.